app/models/concerns/sequenced.rb

```diff
@@ -12,6 +12,7 @@ module Sequenced
 
     def ensure_sequential_id
       return if sequential_id.present?
+      return if self.class.to_s == 'Invoice' && !status_changed_to_finalized?
```

Comment:
Style/ClassEqualityComparison: Use `instance_of?(::Invoice)` instead of comparing classes. (https://rubystyle.guide#instance-of-vs-class-comparison)